src/screens/Manager/Modals/FilterModal.js

```diff
@@ -0,0 +1,254 @@
+import React, { memo, useState, useCallback, useEffect } from "react";
+import {
+  StyleSheet,
+  SectionList,
+  View,
+  TouchableHighlight,
+} from "react-native";
+import { Trans } from "react-i18next";
+import Check from "../../../icons/Check";
+import colors from "../../../colors";
+import LText from "../../../components/LText";
+
+import ActionModal from "./ActionModal";
+
+const filterSections = [
+  {
+    title: "AppAction.filter.title",
+    data: [
+      // {
+      //   label: "AppAction.filter.all",
+      //   value: "all",
+      //   isFilter: true,
+      // },
+      {
+        label: "AppAction.filter.installed",
+        value: "installed",
+        isFilter: true,
+      },
+      {
+        label: "AppAction.filter.not_installed",
+        value: "not_installed",
+        isFilter: true,
+      },
+      {
+        label: "AppAction.filter.supported",
+        value: "supported",
+        isFilter: true,
+      },
+      {
+        label: "AppAction.filter.updatable",
+        value: "updatable",
+        isFilter: true,
+      },
+    ],
+    footerSeparator: true,
+  },
+  {
+    title: "AppAction.sort.title",
+    data: [
+      // {
+      //   label: "AppAction.sort.default",
+      //   value: "default",
+      // },
+      {
+        label: "AppAction.sort.name_asc",
+        value: "name",
+        orderValue: "asc",
+      },
+      {
+        label: "AppAction.sort.name_desc",
+        value: "name",
+        orderValue: "desc",
+      },
+      {
+        label: "AppAction.sort.marketcap_desc",
+        value: "marketcap",
+        orderValue: "desc",
+      },
+      {
+        label: "AppAction.sort.marketcap_asc",
+        value: "marketcap",
+        orderValue: "asc",
+      },
+    ],
+  },
+];
+
+const keyExtractor = (item, index) => item + index;
+
+const SectionHeader = ({ section: { title } }: *) => (
+  <View style={[styles.sectionLine, styles.paddingLine]}>
+    <LText style={styles.sectionHeader}>
+      <Trans i18nKey={title} />
+    </LText>
+  </View>
+);
+
+const Separator = ({ section: { footerSeparator } }) =>
+  Boolean(footerSeparator) && (
+    <View style={styles.paddingLine}>
+      <View style={styles.separator} />
+    </View>
+  );
+
+type Props = {
+  filter: string,
+  setFilter: Function,
+  sort: string,
+  setSort: Function,
+  order: string,
+  setOrder: Function,
+  isOpened: Boolean,
+  onClose: Function,
+};
+
+const FilterModalComponent = ({
+  filter,
+  setFilter,
+  sort,
+  setSort,
+  order,
+  setOrder,
+  isOpened,
+  onClose,
+}: Props) => {
+  const [selectedFilters, filterBy] = useState();
+  const [selectedSort, sortBy] = useState();
+  const [selectedOrder, orderBy] = useState();
+
+  useEffect(() => {
+    filterBy(filter);
+    sortBy(sort);
+    orderBy(order);
```

Comment:
makes me feel we should factorize this into a useReducer to avoid doing 3 re-rendering here.